damus/Models/KFImageModel.swift

```diff
@@ -106,3 +108,28 @@ struct CustomCacheSerializer: CacheSerializer {
         return DefaultCacheSerializer.default.image(with: data, options: options)
     }
 }
+
+class CustomSessionDelegate: SessionDelegate {
```

Comment:
I'm confused as to where the HEAD request is happening here

Reply:
No separate request is being made. The HEAD is being intercepted in the urlSession response which runs before the download starts in KF. I think this is more efficient and cleaner than running a separate HEAD request and then passing the url to KF.

Reply:
interesting...

Reply:
Let me know if you have any other questions or prefer it to be done another way. Also, looks like there are merge conflicts now. I can resolve those once everything looks good.

Reply:
I'm just curious how you know this header is available before the data is downloaded, is there docs for this somewhere?

Reply:
ah awesome